addons/sale_loyalty/models/sale_order.py

```diff
@@ -829,24 +829,21 @@ def _program_check_compute_points(self, programs):
         products_per_rule = programs._get_valid_products(products)
 
         # Prepare amounts
-        no_effect_lines = self._get_no_effect_on_threshold_lines()
-        base_untaxed_amount = self.amount_untaxed - sum(line.price_subtotal for line in no_effect_lines)
-        base_tax_amount = self.amount_tax - sum(line.price_tax for line in no_effect_lines)
-        amounts_per_program = {p: {'untaxed': base_untaxed_amount, 'tax': base_tax_amount} for p in programs}
-        for line in self.order_line:
-            if not line.reward_id or line.reward_id.reward_type != 'discount':
-                continue
+        so_products_per_rule = programs._get_valid_products(self.order_line.product_id)
```

Comment:
Are we sure that eWallets do not decrease the total amount here ?

Reply:
It's already handled correctly :)